scripts/mainPage.js

```diff
@@ -0,0 +1,146 @@
+const marqueeInnerText = document.getElementById("marquee-text");
+const input = document.getElementById("searchLine");
+const searchBtn = document.getElementById("searchBtn");
+const loadingSpinner = document.getElementById("loadingSpinner");
+let baseUrl = `https://stock-exchange-dot-full-stack-course-services.ew.r.appspot.com/api/v3/`;
+
+searchBtn.addEventListener("click", function (event) {
+  event.preventDefault();
+  getDataFromAPI();
+});
+
+// Make a request to the API endpoint for marquee data
+fetch(new URL(`stock-screener?&exchange=NASDAQ`, baseUrl))
```

Comment:
good. put it inside a fuction in order to seperate it from other code and keep the global scope clean.